File: GXpert/GXpert.Web/Modules/Analytics/AssignedExamTeachers/AssignedExamTeachersColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Analytics.Columns;

[ColumnsScript("Analytics.AssignedExamTeachers")]
[BasedOnRow(typeof(AssignedExamTeachersRow), CheckNames = true)]
public class AssignedExamTeachersColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string ExamTitle { get; set; }
    public string PlayListTitle { get; set; }
    public string TeacherPrn { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Analytics/AssignedExamTeachersColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { AssignedExamTeachersRow } from "./AssignedExamTeachersRow";

export interface AssignedExamTeachersColumns {
    Id: Column<AssignedExamTeachersRow>;
    ExamTitle: Column<AssignedExamTeachersRow>;
    PlayListTitle: Column<AssignedExamTeachersRow>;
    TeacherPrn: Column<AssignedExamTeachersRow>;
    InsertDate: Column<AssignedExamTeachersRow>;
    InsertUserId: Column<AssignedExamTeachersRow>;
    UpdateDate: Column<AssignedExamTeachersRow>;
    UpdateUserId: Column<AssignedExamTeachersRow>;
    IsActive: Column<AssignedExamTeachersRow>;
}

export class AssignedExamTeachersColumns extends ColumnsBase<AssignedExamTeachersRow> {
    static readonly columnsKey = 'Analytics.AssignedExamTeachers';
    static readonly Fields = fieldsProxy<AssignedExamTeachersColumns>();
}